scipy/stats/tests/test_stats.py

```diff
@@ -3369,6 +3371,29 @@ def test_1samp_ci_iv(self):
         with pytest.raises(ValueError, match=message):
             res.confidence_interval(confidence_level=10)
 
+    shape = npst.array_shapes(min_dims=1, min_side=2)
+    alpha=strategies.floats(1e-6, 1-1e-6, allow_nan=False)
+    # when there is an error, why does the test take a long time to finish?
+    @given(shape=shape, alpha=alpha)
+    @pytest.mark.parametrize('alternative', ['less', 'greater'])
+    def test_pvalue_ci(self, shape, alpha, alternative):
+        # test the relationship between one-sided p-values and confidence intervals
+        rng = np.random.default_rng(435892655436)
+        # I had a tough time generating this with `npst.arrays` directly?
+        # how do I "pass kwargs through from `arrays()`"
+        # I do run into overflows and other numerical issues for outrageous data -
+        # what do you recommend when the test fails but it's not really the fault
+        # of the function being tested?
+        data = rng.random(size=shape)
+        # can this be generated by hypothesis? It needs to be consistent with `shape`.
+        axis = rng.integers(0, len(data.shape))
+        res = stats.ttest_1samp(data, 0, alternative=alternative, axis=axis)
+        l, u = res.confidence_interval(confidence_level=alpha)
+        popmean = l if alternative == 'greater' else u
+        popmean = np.expand_dims(popmean, axis=axis)
+        res = stats.ttest_1samp(data, popmean, alternative=alternative, axis=axis)
+        np.testing.assert_allclose(res.pvalue, 1 - alpha)
```

Comment:
I advocated a bit for the adoption of `hypothesis` in NumPy a few years ago, so wanted to mention a few things. First, I believe @Zac-HD was helpful in getting NumPy up to speed with `hypothesis` infrastructure in the early stages (as a maintainer of `hypothesis` proper).

> I still need to investigate how we will [reproduce failures](https://hypothesis.readthedocs.io/en/latest/reproducing.html) that occur on CI. If you know, please post.

It probably depends a bit on the type of failure, but for example I intentionally broke your test locally and `hypothesis` just gave me the parameters leading to failure.

```
E   Falsifying example: test_pvalue_ci(
E       self=<scipy.stats.tests.test_stats.TestStudentTest object at 0x7fbd06ad4090>,
E       alternative='greater',
E       shape=(2,),
E       alpha=0.5,
E   )

```

I'm almost certain there's another failure scenario, but it is also dealt with by the docs you linked to, and I've seen it a bunch of times over the years--adding a decorator to reproduce tricky cases:

`You can reproduce this example by temporarily adding @reproduce_failure(..., b'AAAA//AAAAAAAAEA') as a decorator on your test case`

> Is there a way to limit the approximate duration of the test? This takes over a second. That's a lot more time than a property -based test that I might write, even if it's heavily parametrized. What is it doing?

Zac will know better than I, but IIRC test exploration will cycle to extreme values and kind of expand and contract over that space a bit, and it may also be the case that `hypothesis` explores inputs that result in unexpectedly slow performance as well (which may also be useful info). I know there are a bunch of ways to limit where `hypothesis` stands on the spectrum between "quick check of a few cases" and "full blown fuzzer," but approaches like example count tuning  `@settings(max_examples=50)` (instead of default `100`) or the `deadline` in milliseconds, etc. That said, I believe the default settings are intended for regular CI flushing.

I suppose we could `grep` the NumPy code base a bit--I see one test with `@settings(max_examples=10000, deadline=2000)` to control its search expansion for example.

Another useful thing is skipping cases that you might know are very time consuming and/or irrelevant with stuff like `assume(not isnan(x))` if `nan` inputs are problematic/slow but also irrelevant to your test.